bindings/java/src/org/sleuthkit/datamodel/blackboardutils/ArtifactsHelper.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2019 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.datamodel.blackboardutils;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.Blackboard;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import org.sleuthkit.datamodel.SleuthkitCase;
+import org.sleuthkit.datamodel.TskCoreException;
+
+/**
+ * This class helps ingest modules create miscellaneous artifacts.
+ *
+ */
+public final class ArtifactsHelper extends ArtifactHelper {
+
+	private static final Logger logger = Logger.getLogger(ArtifactsHelper.class.getName());
+
+	/**
+	 * Creates an artifact helper for artifacts.
+	 *
+	 * @param caseDb     Sleuthkit case db
+	 * @param moduleName name module using the helper
+	 * @param srcFile    source file
+	 *
+	 */
+	public ArtifactsHelper(SleuthkitCase caseDb, String moduleName, AbstractFile srcFile) {
+		super(caseDb, moduleName, srcFile);
+	}
+
+	/**
+	 * Adds a TSK_GPS_TRACKPOINT artifact.
+	 *
+	 * @param latitude    location latitude, required
+	 * @param longitude   location longitude, required
+	 * @param timeStamp   date/time trackpoint recorded, can be 0 if not
+	 *                    available
+	 * @param name        trackpoint name, can be empty/null if not available
+	 * @param programName name of program that recorded trackpoint , can be
+	 *                    empty or null if not available
+	 *
+	 * @return artifact added
+	 */
+	public BlackboardArtifact addGPSLocation(double latitude, double longitude,
+			long timeStamp, String name, String programName) {
+
+		return addGPSLocation(latitude, longitude, timeStamp, name, programName,
+				Collections.<BlackboardAttribute>emptyList());
```

Comment:
It is not necessary to supply the type parameter. You can just call Collections.emptyList() for greater simplicity. This comment applies to various call sites in this code.